File: zero/Maruko.Zero/Internal/Controllers/SysOperateController.cs
using Maruko.Core.Application;
using Maruko.Core.FreeSql.Internal.AppService;
using Maruko.Core.Web;
using Microsoft.AspNetCore.Cors;
using Microsoft.AspNetCore.Mvc;

namespace Maruko.Zero
{
    [EnableCors("cors")]
    [Route("api/v1/sysOperate/")]
    public class SysOperateController : BaseCurdController<SysOperate, OperateDTO>
    {
        private readonly IOperateService _operate;

        public SysOperateController(ICurdAppService<SysOperate, OperateDTO> curd, IOperateService operate) : base(curd)
        {
            _operate = operate;
        }

        [HttpPost("getMenuOfOperate")]
        public AjaxResponse<object> GetMenuOfOperate(MenuOfOperateRequest request)
        {
            return new AjaxResponse<object>(_operate.GetMenuOfOperate(request), "ok");
        }

        [HttpPost("getMenuOfOperateByRole")]
        public AjaxResponse<object> GetMenuOfOperateByRole(GetMenuOfOperateByRoleRequest request)
        {
            return new AjaxResponse<object>(_operate.GetMenuOfOperateByRole(request), "ok");
        }
    }
}

File: act/act-modules/act-modules-zero/src/main/java/com/act/modules/zero/internal/controllers/SysOperateController.java
package com.act.modules.zero.internal.controllers;

import com.act.core.utils.AjaxResponse;
import com.act.core.web.BaseController;
import com.act.modules.zero.internal.application.operate.SysOperateService;
import com.act.modules.zero.internal.application.operate.dto.GetMenuOfOperateByRoleRequest;
import com.act.modules.zero.internal.application.operate.dto.MenuOfOperateRequest;
import com.act.modules.zero.internal.application.operate.dto.SysOperateDTO;
import com.act.modules.zero.internal.domain.SysOperate;
import com.act.modules.zero.internal.mapper.SysOperateMapper;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Api(tags = "系统功能")
@RestController
@RequestMapping("/api/v1/sysOperate/")
@SuppressWarnings("all")
public class SysOperateController extends BaseController<SysOperate, SysOperateDTO, SysOperateMapper> {

    @Autowired
    private SysOperateService _operate;

    @ApiOperation(value = "获取菜单的功能")
    @PostMapping("getMenuOfOperate")
    public AjaxResponse<Object> getMenuOfOperate(@RequestBody MenuOfOperateRequest request) {
        return new AjaxResponse<Object>(_operate.getMenuOfOperate(request));
    }

    @ApiOperation(value = "获取角色下菜单的功能")
    @PostMapping("getMenuOfOperateByRole")
    public AjaxResponse<Object> GetMenuOfOperateByRole(@RequestBody GetMenuOfOperateByRoleRequest request) {
        return new AjaxResponse<Object>(_operate.getMenuOfOperateByRole(request));
    }
}
